sahha-android/src/main/java/sdk/sahha/android/domain/interaction/SahhaInteractionManager.kt

```diff
@@ -74,7 +75,11 @@ internal class SahhaInteractionManager @Inject constructor(
 
                 permission.startHcOrNativeDataCollection(application) { error, successful ->
                     if (permission.manager.shouldUseHealthConnect())
-                        scheduleInsightsAlarm(application)
+                        permission.manager.getHealthConnectSensorStatus { status ->
+                            if (status == SahhaSensorStatus.enabled)
+                                scheduleInsightsAlarm(application)
```

Comment:
Might have to remove this check - we may end up missing some data.

Update insights post logic to check permission during posting and post as long as `shouldUseHealthConnect` is `true`